src/components/shared/sidebar/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import { useMutation } from "react-query";
+import { HomeIcon, InboxIcon, PlusCircleIcon } from "@heroicons/react/24/outline";
+import Router from "next/router";
+
+import { logout } from "@/pages/api";
+import { useStore } from "@/pages/store/store";
+import NavItem from "./NavItem";
+
+export default function Sidebar() {
+  const { token, user } = useStore();
+  const [selected, setSelected] = useState("Menu");
+
+  const mutationLogout = useMutation(() => logout(token), {
+    onSuccess: () => {
+      useStore.getState().setToken("");
+      useStore.getState().setUser(undefined);
+      Router.push("/auth");
+    },
+  });
+
+  return (
+    <>
+      <div className="fixed inset-y-0 z-50 flex w-72 flex-col">
+        <div className="flex grow flex-col gap-y-5 overflow-y-auto border-r border-gray-200 bg-gray-800 px-6 text-gray-300">
+          <nav className="flex flex-1 flex-col pt-10">
+            <ul role="list" className="flex flex-1 flex-col gap-y-2">
+              <NavItem
+                name="Menu"
+                href="/#"
+                navIcon={HomeIcon}
+                selected={selected === "Menu"}
+                onClick={() => setSelected("Menu")}
+              />
+              {user && user.role.includes("PATIENT") && (
+                <NavItem
+                  name="New Submission"
+                  href="/#"
+                  navIcon={PlusCircleIcon}
+                  selected={selected === "New Submission"}
+                  onClick={() => setSelected("New Submission")}
+                />
+              )}
+              {user && user.role.includes("DOCTOR") && (
+                <NavItem
+                  name="Task History"
+                  href="/#"
+                  navIcon={InboxIcon}
+                  selected={selected === "Task History"}
+                  onClick={() => setSelected("Task History")}
+                />
+              )}
+
+              <li className="-mx-6 mt-auto bg-gray-700 text-white">
+                <div className="flex items-center gap-x-4 px-6 py-3 text-sm font-semibold leading-6">
+                  <div className="relative flex h-12 w-12 items-center justify-center rounded-full bg-gray-400 p-3">
+                    <span>{user?.name.charAt(0).toUpperCase()}</span>
+                  </div>
+                  <div className="flex flex-col">
+                    <span aria-hidden="true">{user?.name}</span>
+                    <span
+                      className="cursor-pointer text-xs text-gray-300"
+                      onClick={() => mutationLogout.mutate()}
```

Comment:
here you then pass the token as I've mentioned in the other comment

```suggestion
                      onClick={() => mutationLogout.mutate(token)}
```